### PR TITLE
Don't quote the addresses when we hit errors setting interface addresses

### PR DESCRIPTION
**Pull request by nalind**
Before we call setInterfaceIP() and setInterfaceIPv6(), we construct partial error messages for use if they return errors.  While both functions check for nil addresses, we don't guard against passing nil
addresses to fmt.Sprintf when constructing those messages, which causes us to panic if they're nil.  Add wrappers that supply the text "(no address)" when the address is nil.

### CODE REVIEW

File: osl/interface_linux.go [hunk 1]
```diff
@@ -307,14 +307,26 @@ func (n *networkNamespace) AddInterface(srcName, dstPrefix string, options ...If
...
+		}
+		return fmt.Sprintf("%q", i.AddressIPv6())
+	}
 	ifaceConfigurators := []struct {
 		Fn         func(netlink.Link, *nwIface) error
 		ErrMessage string
 	}{
 		{setInterfaceName, fmt.Sprintf("error renaming interface %q to %q", ifaceName, i.DstName())},
 		{setInterfaceMAC, fmt.Sprintf("error setting interface %q MAC to %q", ifaceName, i.MacAddress())},
-		{setInterfaceIP, fmt.Sprintf("error setting interface %q IP to %q", ifaceName, i.Address())},
```
Comment (aboch) [inline osl/interface_linux.go]: Can you please just change the Sprintf for the set ip functions to
 `interface %q IP to %v"`

Comment (nalind) [inline osl/interface_linux.go]: Hmm, doing that instead doesn't prevent the segfault.
I looked a little closer, and I was incorrect about the effect of this segfault: the runtime catches it (even in the original version), notices that the segfault happened while calling Error() or String() on a nil pointer, ignores the segfault because it special-cases it, and produces "<nil>" in the output message.  I'm attached with a debugger, so I can continue past these when they crop up, but I wouldn't have guessed that it was supposed to work like that.
Long story short, we can either patch to avoid passing the pointer without checking it first, or we can let the runtime keep doing its thing and close this without merging.

Comment (aboch) [inline osl/interface_linux.go]: > the runtime catches it (even in the original version),

Yes I was wondering about that. Because `setInterfaceIPv6()` is called pretty frequently with a nil address.

I am fine to change it to `%v`.

Comment (nalind) [inline osl/interface_linux.go]: Sure, I'll switch it to that.

**Comment by aboch** [comment]
LGTM

**Comment by mavenugo** [comment]
LGTM
